Lib/test/test_site.py

```diff
@@ -465,6 +465,59 @@ def test_sitecustomize_executed(self):
             else:
                 self.fail("sitecustomize not imported automatically")
 
+    @support.requires_subprocess()
+    def test_customization_modules_on_startup(self):
+        # Check that sitecustomize and or usercustomize are executed on startup
+        mod_info = [
+            # func to get directory, file base name
+            ('getsitepackages', 'sitecustomize')
+        ]
+
+        if site.ENABLE_USER_SITE:
+            mod_info.append(('getusersitepackages', 'usercustomize'),)
+
+        for func_name, module_name in mod_info:
+            # getusersitepackages returns a string.. getsitepackages returns a list..
+            # handle either way.
+            base_path = getattr(site, func_name)()
+            if not isinstance(base_path, str):
+                base_path = base_path[0]
+
+            try:
+                os.makedirs(base_path, exist_ok=True)
+            except PermissionError:
+                # Can't modify system site packages depending on the system configuration
+                continue
+
+            customize_path = os.path.join(base_path, f'{module_name}.py')
+            if os.path.exists(customize_path):
+                # backup old sitecustomize.py
+                oldcustomize_path = customize_path + '.old'
+                if os.path.exists(oldcustomize_path):
```

Comment:
I don't know that I love this, could we instead write a sitecustomize into a temp dir, and then put that dir into `PYTHONPATH` env var of the subprocess?